### ISSUE TITLE
BUG: matplotlib as hard dependency

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#2227]
from https://ci.appveyor.com/project/josef-pkt/statsmodels/build/1.0.331

```
======================================================================
ERROR: statsmodels.genmod.tests.test_gee.test_plots
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\envs\py3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\genmod\tests\test_gee.py", line 1369, in test_plots
    import matplotlib.pyplot as plt
ImportError: No module named 'matplotlib'
 
======================================================================
ERROR: statsmodels.genmod.tests.test_glm.test_plots
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\envs\py3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\genmod\tests\test_glm.py", line 815, in test_plots
    import matplotlib.pyplot as plt
ImportError: No module named 'matplotlib'
 
======================================================================
ERROR: statsmodels.sandbox.tests.test_predict_functional.TestPredFunc.test_glm_formula
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\envs\py3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\sandbox\tests\test_predict_functional.py", line 264, in test_glm_formula
    plt.clf()
NameError: name 'plt' is not defined
 
======================================================================
ERROR: statsmodels.sandbox.tests.test_predict_functional.TestPredFunc.test_glm_formula_contrast
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\envs\py3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\sandbox\tests\test_predict_functional.py", line 151, in test_glm_formula_contrast
    plt.clf()
NameError: name 'plt' is not defined
 
======================================================================
ERROR: statsmodels.sandbox.tests.test_predict_functional.TestPredFunc.test_lm_contrast
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\envs\py3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\sandbox\tests\test_predict_functional.py", line 115, in test_lm_contrast
    plt.clf()
NameError: name 'plt' is not defined
 
======================================================================
ERROR: statsmodels.sandbox.tests.test_predict_functional.TestPredFunc.test_noformula_prediction
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\envs\py3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\sandbox\tests\test_predict_functional.py", line 320, in test_noformula_prediction
    plt.clf()
NameError: name 'plt' is not defined
 
======================================================================
ERROR: statsmodels.sandbox.tests.test_predict_functional.TestPredFunc.test_scb
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\envs\py3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\sandbox\tests\test_predict_functional.py", line 207, in test_scb
    plt.clf()
NameError: name 'plt' is not defined
 
======================================================================
ERROR: Failure: ImportError (No module named 'matplotlib')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\envs\py3\lib\site-packages\nose\failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "C:\Python27\envs\py3\lib\site-packages\nose\loader.py", line 414, in loadTestsFromName
    addr.filename, addr.module)
  File "C:\Python27\envs\py3\lib\site-packages\nose\importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "C:\Python27\envs\py3\lib\site-packages\nose\importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "C:\Python27\envs\py3\lib\imp.py", line 235, in load_module
    return load_source(name, filename, file)
  File "C:\Python27\envs\py3\lib\imp.py", line 171, in load_source
    module = methods.load()
  File "<frozen importlib._bootstrap>", line 1220, in load
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\tools\tests\test_pca.py", line 10, in <module>
    from statsmodels.tools.pca import PCA
  File "C:\Python27\envs\py3\lib\site-packages\statsmodels-0.7.0-py3.4-win32.egg\statsmodels\tools\pca.py", line 9, in <module>
    import matplotlib.pyplot as plt
ImportError: No module named 'matplotlib'
```

**Comment by josef-pkt** [comment]
@kshedden  unprotected use of matplotlib, all except one are yours.
Can you put a try..except import of matplotlib as we do in `graphics`?

**Comment by bashtage** [comment]
Should there be a Travis test w/o matplotlib, since it is optional?

**Comment by josef-pkt** [comment]
@bashtage 
We should have one test version without matplotlib. But if I activate appveyor soon, then we have one without matplotlib. Because of the time limitation, we have to restrict what we run on appveyor.

If you think we can drop one matplotlib from Travis without loosing much in terms of version check, then you could do it for now.

**Comment by josef-pkt** [comment]
This has been fixed by #2229 #2230 #2245

travis is now using one python version without matplotlib to avoid these errors PR #2246
